src/schemas/json/appsettings.json

```diff
@@ -90,16 +91,241 @@
+        },
+        "Subject": {
+          "description": "The certificate subject. If a subject is specified then the certificate will be loaded from the certificate store.",
+          "type": "string"
+        },
+        "Store": {
+          "description": "The certificate store name. Defaults to 'My'.",
+          "type": "string",
+          "default": "My"
+        },
+        "Location": {
+          "description": "The certificate store location. Defaults to 'CurrentUser'.",
+          "type": "string",
+          "enum": [
+            "LocalMachine",
+            "CurrentUser"
+          ],
+          "default": "CurrentUser"
+        },
+        "AllowInvalid": {
+          "description": "A value indicating whether or not to load certificates that are considered invalid. Defaults to false.",
+          "type": "boolean",
+          "default": false
+        }
+      }
+    },
+    "sslProtocols": {
+      "description": "Specifies allowable SSL protocols. Defaults to 'None' which allows the operating system to choose the best protocol to use, and to block protocols that are not secure. Unless your app has a specific reason not to, you should use this default.",
+      "type": "array",
+      "items": {
+        "type": "string",
+        "enum": [
+          "None",
+          "Ssl2",
+          "Ssl3",
+          "Tls",
+          "Default",
+          "Tls11",
+          "Tls12",
+          "Tls13"
+        ],
+        "default": "None"
+      }
+    },
+    "clientCertificateMode": {
+      "description": "Specifies the client certificate requirements for a HTTPS connection. Defaults to 'NoCertificate'.",
+      "type": "string",
+      "enum": [
+        "NoCertificate",
+        "AllowCertificate",
+        "RequireCertificate"
+      ],
+      "default": "NoCertificate"
+    },
+    "kestrel": {
+      "type": "object",
+      "description": "Configuration of Kestrel-specific features.",
+      "properties": {
+        "Endpoints": {
+          "description": "Endpoints that Kestrel listens to for network requests. Each endpoint has a name specified by its JSON property name.",
+          "type": "object",
+          "additionalProperties": {
+            "description": "Kestrel endpoint configuration.",
+            "type": "object",
+            "properties": {
+              "Url": {
+                "description": "The host name and port the endpoint will listen on. A Url is required.",
+                "type": "string",
+                "format": "uri"
+              },
+              "Protocols": {
+                "$ref": "#/definitions/protocols"
+              },
+              "SslProtocols": {
+                "$ref": "#/definitions/sslProtocols"
+              },
+              "Certificate": {
+                "$ref": "#/definitions/certificate"
+              },
+              "ClientCertificateMode": {
+                "$ref": "#/definitions/clientCertificateMode"
+              },
+              "Sni": {
```

Comment:
How does this schema handle versioning? E.g. some of these properties like SNI are new in 5.0.

Reply:
There is only one schema so everything will show up. The description can mention that a setting is new in .NET 5.

Is Sni the only property that is new?

Reply:
New/Changed:
- ClientCertificateMode: https://github.com/dotnet/aspnetcore/issues/26246
- SslProtocols default changed: https://github.com/dotnet/aspnetcore/issues/26244
- SNI: https://github.com/dotnet/aspnetcore/issues/26242
- Certificate KeyPath: https://github.com/dotnet/aspnetcore/pull/23584

Reply:
Kestrel:EndpointDefaults:SslProtocols and Kestrel:EndpointDefaults:ClientCertificateMode are also new in .NET 5.